python/cuml/solvers/qn.pyx

```diff
@@ -267,6 +280,7 @@ class QN(Base,
         self.l2_strength = l2_strength
         self.max_iter = max_iter
         self.tol = tol
+        self.delta = delta if delta is not None else (tol * 0.5)
```

Comment:
I will also need to update `get_param_names`, but it will conflict with #3774 . Wait till that one is merged.